frontend/src/components/ProgressIndicator/ProgressIndicator.tsx

```diff
@@ -1,35 +1,55 @@
 import { useMemo } from 'react'
-import { Box } from '@chakra-ui/react'
+import { Box, BoxProps } from '@chakra-ui/react'
+import { getColor } from '@chakra-ui/theme-tools'
 
+import { theme } from '~theme/index'
 import { MotionBox } from '~components/motion'
 
-const ActiveIndicator = (): JSX.Element => (
+const ActiveIndicator = ({ ...props }: BoxProps): JSX.Element => (
   <Box
+    // Top required to align it with CircleIndicators
+    top="0.125rem"
     width="1.5rem"
     height="0.5rem"
     borderRadius="full"
     backgroundColor="secondary.500"
+    position="absolute"
+    as="button"
+    {...props}
   />
 )
 
-interface CircleIndicatorProps {
+interface CircleIndicatorProps extends BoxProps {
   onClick: () => void
   isActiveIndicator: boolean
 }
 
 const CircleIndicator = ({
   onClick,
   isActiveIndicator,
+  ...props
 }: CircleIndicatorProps): JSX.Element => {
   return (
     <Box
-      width="0.5rem"
-      height="0.5rem"
+      width="0.75rem"
+      height="0.75rem"
+      padding="0.125rem"
       borderRadius="full"
       backgroundColor="secondary.200"
-      marginRight={isActiveIndicator ? '1.5rem' : '0.5rem'}
+      marginRight={isActiveIndicator ? '1.25rem' : '0.25rem'}
       onClick={onClick}
+      _hover={{ backgroundColor: 'secondary.300' }}
+      _focus={
+        isActiveIndicator
+          ? undefined
+          : {
+              backgroundColor: 'secondary.300',
+              boxShadow: `0 0 0 1px ${getColor(theme, 'secondary.400')}`,
```

Comment:
instead of function call, we can use css variables:

```tsx
boxShadow: '0 0 0 1px var(--chakra-colors-secondary-400)'
```

read more about the ones chakra generates app-wide here: https://chakra-ui.com/docs/styled-system/features/css-variables